python/shotgun_menus/entity_field_menu.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+This module implements a QMenu subclass that knows how to display all the fields
+for a given Shotgun entity type.
+
+An example of how to use it is:
+
+    class AppDialog(QtGui.QWidget):
+        def __init__(self):
+            QtGui.QWidget.__init__(self)
+
+            # grab a field manager to know what fields are displayable
+            self._field_manager = shotgun_fields.ShotgunFieldManager()
+
+            # setup a label to have the fields menu as its context menu
+            self.label = QtGui.QLabel("Right click me!")
+            self.label.setContextMenuPolicy(QtCore.Qt.CustomContextMenu)
+            self.label.customContextMenuRequested.connect(self.open_menu)
+
+            # and layout the dialog
+            layout = QtGui.QVBoxLayout(self)
+            layout.addWidget(self.label)
+            self.setLayout(layout)
+
+        def field_filter(self, field):
+            # display fields that are displayable by the shotgun field widgets
+            return bool(self._field_manager.supported_fields("CustomEntity02", [field]))
+
+        def checked_filter(self, field):
+            # none of the fields are checked
+            return False
+
+        def disabled_filter(self, field):
+            # none of the fields are disabled
+            return False
+
+        def open_menu(self, position):
+            menu = shotgun_menus.EntityFieldMenu("CustomEntity02")
+
+            # attach our filters
+            menu.set_field_filter(self.field_filter)
+            menu.set_checked_filter(self.checked_filter)
+            menu.set_disabled_filter(self.disabled_filter)
+
+            # show the menu and print the result
+            action = menu.exec_(self.label.mapToGlobal(position))
+            if action:
+                # action's data has the field that was selected
+                self.do_thing(action.data()["field"])
+"""
+import sgtk
+from sgtk.platform.qt import QtGui
+
+shotgun_globals = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_globals")
+
+
+class EntityFieldMenu(QtGui.QMenu):
+    """
+    Inherited from a :class:`~PySide.QtGui.QMenu`, this menu will automatically display the fields
+    for a given Shotgun entity.
+
+    The QActions for the menu will all have their data set to a dictionary in the form:
+        {"field": selected_field}
+    """
+    _AUDIT_FIELDS = ["created_by", "created_at", "updated_by", "updated_at"]
```

Comment:
@manneohrstrom do you have a use case in mind?  Easy enough to move there if needed, but unless you have something in mind I'd rather wait than create a dependency between this and a core update.